android/apollo/src/com/andrew/apollo/utils/ApolloUtils.java

```diff
@@ -144,47 +145,66 @@ public static final ImageFetcher getImageFetcher(final Activity activity) {
-            intent.putExtra(Intent.EXTRA_SHORTCUT_ICON, BitmapUtils.resizeAndCropCenter(bitmap, 96));
-            intent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, shortcutIntent);
-            intent.putExtra(Intent.EXTRA_SHORTCUT_NAME, displayName);
-            intent.setAction("com.android.launcher.action.INSTALL_SHORTCUT");
-            context.sendBroadcast(intent);
-            AppMsg.makeText(context,
-                    context.getString(R.string.pinned_to_home_screen, displayName),
-                    AppMsg.STYLE_CONFIRM).show();
-        } catch (final Exception e) {
-            Log.e("ApolloUtils", "createShortcutIntent", e);
-            AppMsg.makeText(
-                    context,
-                    context.getString(R.string.could_not_be_pinned_to_home_screen, displayName),
-                    AppMsg.STYLE_ALERT).show();
-        }
+                    final Intent shortcutIntent = new Intent(context, ShortcutActivity.class);
+                    shortcutIntent.setAction(Intent.ACTION_VIEW);
+                    shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                    shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
+                    shortcutIntent.putExtra(Config.ID, id);
+                    shortcutIntent.putExtra(Config.NAME, displayName);
+                    shortcutIntent.putExtra(Config.MIME_TYPE, mimeType);
+
+                    // Intent that actually sets the shortcut
+                    final Intent intent = new Intent();
+                    intent.putExtra(Intent.EXTRA_SHORTCUT_ICON, BitmapUtils.resizeAndCropCenter(bitmap, 96));
+                    intent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, shortcutIntent);
+                    intent.putExtra(Intent.EXTRA_SHORTCUT_NAME, displayName);
+                    intent.setAction("com.android.launcher.action.INSTALL_SHORTCUT");
+                    context.sendBroadcast(intent);
+                } catch (Exception e) {
+                    Log.e("ApolloUtils", "createShortcutIntent", e);
+                    success = false;
+                }
+
+                final boolean finalSuccess = success;
+
+                // UI thread portion
+                Runnable postExecute = new Runnable() {
+                    public void run() {
+                        if (finalSuccess) {
+                            AppMsg.makeText(context,
+                                    context.getString(R.string.pinned_to_home_screen, displayName),
+                                    AppMsg.STYLE_CONFIRM).show();
+                        } else {
+                            AppMsg.makeText(
+                                    context,
+                                    context.getString(R.string.could_not_be_pinned_to_home_screen, displayName),
+                                    AppMsg.STYLE_ALERT).show();
+                        }
+                    }
+                };
+                context.runOnUiThread(postExecute);
```

Comment:
Well, this will hold it alive, right?
I tested that with Thread.sleep(10000) to simulate a long running image get. 
Backed out of the activity/ home buttoned and locked and no problems.